autokeras/hypermodel/hyper_block.py

```diff
@@ -102,40 +96,91 @@ def build(self, hp, inputs=None):
         input_node = tf.reshape(input_node, [-1, shape[1], feature_size])
         output_node = input_node
 
-        in_layer = layer_utils.get_rnn_block(
-            hp.Choice('rnn_type', ['vanilla', 'gru', 'lstm'], default='vanilla'))
+        in_layer = const.Constant.RNN_LAYERS[hp.Choice('rnn_type',
+                                                       ['gru', 'lstm'],
+                                                       default='lstm')]
         choice_of_layers = hp.Choice('num_layers', [1, 2, 3], default=2)
 
-        for i in range(choice_of_layers):
-            return_sequences = False if i == choice_of_layers - 1 else True
-            bidirectional_block = tf.keras.layers.Bidirectional(
-                in_layer(feature_size, activation='tanh',
-                         return_sequences=return_sequences))
-            output_node = bidirectional_block(output_node)
+        bidirectional = self.bidirectional
+        if bidirectional is None:
+            bidirectional = hp.Choice('bidirectional',
+                                      [True, False],
+                                      default=True)
 
-        output_node = Flatten().build(hp, output_node)
+        return_sequences = self.return_sequences
+        if return_sequences is None:
+            return_sequences = hp.Choice('return_sequences',
+                                         [True, False],
+                                         default=True)
+
+        for i in range(choice_of_layers):
+            temp_return_sequences = True
+            if i == choice_of_layers - 1:
+                temp_return_sequences = return_sequences
+            if bidirectional:
+                output_node = tf.keras.layers.Bidirectional(
+                    in_layer(feature_size,
+                             return_sequences=temp_return_sequences))(output_node)
+            else:
+                output_node = in_layer(feature_size,
+                                       return_sequences=return_sequences)
 
         return output_node
 
 
 class ImageBlock(HyperBlock):
 
     def build(self, hp, inputs=None):
-        # TODO: make it more advanced, selecting from multiple models, e.g.,
-        #  ResNet.
         input_node = layer_utils.format_inputs(inputs, self.name, num=1)[0]
         output_node = input_node
 
-        for i in range(hp.Choice('num_layers', [1, 2, 3], default=2)):
+        block_type = hp.Choice('block_type',
+                               ['resnet', 'xception', 'vanilla'],
+                               default='resnet')
+
+        if block_type == 'resnet':
+            output_node = ResNetBlock().build(hp, output_node)
+        elif block_type == 'xception':
+            output_node = XceptionBlock().build(hp, output_node)
+        elif block_type == 'vanilla':
+            output_node = ConvBlock().build(hp, output_node)
```

Comment:
I prefer having a `string` to `Block` map `block_map` then just do `block_map[block_type].build(hp, output_node)`